drivers/amazonec2/amazonec2.go

```diff
@@ -1546,6 +1548,25 @@ func (d *Driver) getRegionZone() string {
 	return d.Zone
 }
 
+// buildEC2Tags accepts a string of tagGroups (in the format of 'key1,value1,key2,value2')
+// and returns a slice of ec2.Tag's which can be applied to various ec2 resources.
+func buildEC2Tags(tagGroups string) []*ec2.Tag {
+	tags := make([]*ec2.Tag, 0, len(tagGroups)/2)
```

Comment:
I don't think this works: `tagGroups` is a string, so the length of that string is not what you want. I think this has to go back to the way it was and use my suggestion below.